### PR TITLE
Fix new node monitoring configuration

### PR DESCRIPTION
**Pull request by anmolbabu**
tendrl-bug-id: Tendrl/performance-monitoring#50
Signed-off-by: anmolbabu <anmolbudugutta@gmail.com>

### CODE REVIEW

**Comment by anmolbabu** [comment]
@shtripat Yes the etcd watch worked from main thread only and I had issues with it working in a process spawned from main thread. So what @nthomas-redhat suggested is, to try spawning a gevent for watch only, so that even if it is blocking, the rest of the job is anyway each of them done as a multiprocess soa blocked gevent wouldn't harm and watch worked with this approach and again with this approach I had to also bring back, https://github.com/Tendrl/performance-monitoring/pull/54/files#diff-feb190459407f84b9db76609e06a3c6fR71 :)

**Comment by anmolbabu** [comment]
@Tendrl/tendrl-core Please review

**Comment by shtripat** [comment]
LGTM otherwise

File: tendrl/performance_monitoring/configure/configure_monitoring.py [hunk 1]
```diff
@@ -58,11 +35,32 @@ def init_monitoring(self):
...
-            super(ConfigureMonitoring, self).__init__()
             self.init_monitoring()
         except TendrlPerformanceMonitoringException as ex:
             raise ex
 
-    def run(self):
-        self.watch_nodes()
+    def _run(self):
+        try:
+            while True:
```
Comment (r0h4n) [inline tendrl/performance_monitoring/configure/configure_monitoring.py]: https://github.com/Tendrl/gluster-integration/blob/master/tendrl/gluster_integration/sds_sync/__init__.py#L27

update as per above, while true is not ideal

File: tendrl/performance_monitoring/configure/configure_monitoring.py [hunk 2]
```diff
@@ -58,11 +35,36 @@ def init_monitoring(self):
...
+            self._complete = gevent.event.Event()
         except TendrlPerformanceMonitoringException as ex:
             raise ex
 
-    def run(self):
-        self.watch_nodes()
+    def _run(self):
+        try:
+            while not self._complete.is_set():
+                node_changes = self.etcd_orm.client.watch(
```
Comment (r0h4n) [inline tendrl/performance_monitoring/configure/configure_monitoring.py]: you should make this loop yield at regular intervals

please add a "gevent.sleep(1)" right after L47

**Pull request merged**